### PR TITLE
Make docker gitea/gitea:v1.16-dev etc refer to the latest build on that branch

### PR DESCRIPTION
**Pull request by zeripath**
One of the problems with our current docker tagging is that although we
have strict version tags, latest and dev we do not have a way for docker
users to track the current release branch. This PR simply suggests that
we use the 1.x-dev tag for these and we build and push these. This will
give users who want or need unreleased bug fixes the option of tracking
the pre-release version instead of simply jumping to dev.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by 6543 — APPROVED** [review]
ACK

**Comment by lunny** [comment]
That should be 1.16 but not 1.16-dev ?

**Comment by 6543** [comment]
1.16 track latest specific release tag ... 1.16-dev track latest commit to release branch

**Comment by lunny** [comment]
> 1.16 track latest specific release tag ... 1.16-dev track latest commit to release branch

So we really need some documents update.

**Comment by zeripath** [comment]
As it stands currently - with  "auto tagging" pushing tags to docker 1.x.y will auto tag 1.x and 1 too. (auto tagging as defined by https://github.com/drone-plugins/drone-manifest/blob/92722b712a4a378262ca8f3e39e34248646c8a7b/tagging/tagging.go#L21)

Thus gitea/gitea:1 = gitea/gitea:1.16 = gitea/gitea:1.16.0

and 

gitea/gitea:1.15 = gitea/gitea:1.15.11

If we were to push the branch to 1.16 without the dev suffix we'd be changing that.

**Comment by zeripath** [comment]
I've pushed up some non-automatic specific documentation changes but I can make it more generic if we're gonna backport this to 1.15.

**Comment by zeripath** [comment]
A more generic phrase would be: 

If you'd like to run the latest commit from a release branch you can use the `:1.x-dev` tag, where x is the current minor version of gitea. (e.g. `:1.16-dev`)

**Review by 6543 — APPROVED** [review]

File: docs/content/doc/installation/with-docker-rootless.en-us.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ image as a service. Since there is no database available, one can be initialized
 Create a directory for `data` and `config` then paste the following content into a file named `docker-compose.yml`.
 Note that the volume should be owned by the user/group with the UID/GID specified in the config file. By default Gitea in docker will use uid:1000 gid:1000. If needed you can set ownership on those folders with the command: `sudo chown 1000:1000 config/ data/`
 If you don't give the volume correct permissions, the container may not start.
-For a stable release you could use `:latest-rootless`, `:1-rootless` or specify a certain release like `:{{< version >}}-rootless`, but if you'd like to use the latest development version then `:dev-rootless` would be an appropriate tag.
+For a stable release you could use `:latest-rootless`, `:1-rootless` or specify a certain release like `:{{< version >}}-rootless`, but if you'd like to use the latest development version then `:dev-rootless` would be an appropriate tag. If you'd like to run the latest commit from the 1.16 release branch you can use the `:1.16-dev-rootless` tag.
```
Comment (zeripath) [inline docs/content/doc/installation/with-docker-rootless.en-us.md]:
```suggestion
For a stable release you could use `:latest-rootless`, `:1-rootless` or specify a certain release like `:{{< version >}}-rootless`, but if you'd like to use the latest development version then `:dev-rootless` would be an appropriate tag. If you'd like to run the latest commit from a release branch you can use the `:1.x-dev-rootless` tag, where x is the minor version of Gitea. (e.g. `:1.16-dev`)
```

File: docs/content/doc/installation/with-docker.en-us.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ image as a service. Since there is no database available, one can be initialized
 Create a directory like `gitea` and paste the following content into a file named `docker-compose.yml`.
 Note that the volume should be owned by the user/group with the UID/GID specified in the config file.
 If you don't give the volume correct permissions, the container may not start.
-For a stable release you can use `:latest`, `:1` or specify a certain release like `:{{< version >}}`, but if you'd like to use the latest development version of Gitea then you could use the `:dev` tag.
+For a stable release you can use `:latest`, `:1` or specify a certain release like `:{{< version >}}`, but if you'd like to use the latest development version of Gitea then you could use the `:dev` tag. If you'd like to run the latest commit from the 1.16 release branch you can use the `:1.16-dev` tag.
```
Comment (zeripath) [inline docs/content/doc/installation/with-docker.en-us.md]:
```suggestion
For a stable release you can use `:latest`, `:1` or specify a certain release like `:{{< version >}}`, but if you'd like to use the latest development version of Gitea then you could use the `:dev` tag. If you'd like to run the latest commit from a release branch you can use the `:1.x-dev` tag, where x is the minor version of Gitea. (e.g. `:1.16-dev`)
```

**Review by wxiaoguang — APPROVED** [review]

File: docs/content/doc/installation/with-docker-rootless.en-us.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ image as a service. Since there is no database available, one can be initialized
 Create a directory for `data` and `config` then paste the following content into a file named `docker-compose.yml`.
 Note that the volume should be owned by the user/group with the UID/GID specified in the config file. By default Gitea in docker will use uid:1000 gid:1000. If needed you can set ownership on those folders with the command: `sudo chown 1000:1000 config/ data/`
 If you don't give the volume correct permissions, the container may not start.
-For a stable release you could use `:latest-rootless`, `:1-rootless` or specify a certain release like `:{{< version >}}-rootless`, but if you'd like to use the latest development version then `:dev-rootless` would be an appropriate tag.
+For a stable release you could use `:latest-rootless`, `:1-rootless` or specify a certain release like `:{{< version >}}-rootless`, but if you'd like to use the latest development version then `:dev-rootless` would be an appropriate tag. If you'd like to run the latest commit from a release branch you can use the `:1.x-dev-rootless` tag, where x is the minor version of Gitea. (e.g. `:1.16-dev`)
```
Comment (KN4CK3R) [inline docs/content/doc/installation/with-docker-rootless.en-us.md]:
```suggestion
For a stable release you could use `:latest-rootless`, `:1-rootless` or specify a certain release like `:{{< version >}}-rootless`, but if you'd like to use the latest development version then `:dev-rootless` would be an appropriate tag. If you'd like to run the latest commit from a release branch you can use the `:1.x-dev-rootless` tag, where x is the minor version of Gitea. (e.g. `:1.16-dev-rootless`)
```

**Comment by 6543** [comment]
:rocket:

**Pull request merged**